### PR TITLE
Send product name to updates-server

### PR DESCRIPTION
**Pull request by oguzeroglu**
Related to https://github.com/camunda/camunda-modeler-update-server/issues/36

__Acceptance Criteria__

* [x] Corresponds to the concept 
* [x] Corresponds to the design 

__Definition of Done__

* [x] corresponds to [the design principles](https://github.com/bpmn-io/design-principles)
* [x] corresponds to [the code standards](https://github.com/bpmn-io/bpmn-js/blob/master/.github/CONTRIBUTING.md#creating-a-pull-request)
* [x] passes Continuous Integration checks
* [x] available as feature branch on GitHub
  * [x] contains the cleaned up commit history
  * [x] commit messages satisfy our [commit message guidelines](https://www.conventionalcommits.org/)
  * [x] a single commit closes the issue via Closes #issuenr

### CODE REVIEW

File: client/src/plugins/update-checks/UpdateChecksAPI.js [hunk 1]
```diff
@@ -15,6 +15,8 @@ const OS_INFO_CONFIG_KEY = 'os.info';
 
 const UPDATE_CHECK_API = 'update-check';
 
+const PRODUCT_CAMUNDA_MODELER = 'Camunda Modeler';
```
Comment (barmac) [inline client/src/plugins/update-checks/UpdateChecksAPI.js:18]: I'd love the product name to be set on build time. Thus, we won't have to change anything in the code when we implement this feature in Zeebe Modeler.

Comment (oguzeroglu) [reply client/src/plugins/update-checks/UpdateChecksAPI.js:18]: 👍

File: client/src/plugins/update-checks/__tests__/UpdateChecksSpec.js [hunk 2]
```diff
@@ -295,6 +295,36 @@ describe('<UpdateChecks>', function() {
...
+        downloadURL: 'test-download-url',
+        releases: []
+      }
+    }, (url) => { calledURL = url; });
+
+    // when
+    await tick(component);
+
+    // then
+    expect(calledURL).to.eql('http://test-update-server.com/update-check?editorID=test-id&newerThan=v3.5.0&modelerVersion=v3.5.0&os=windows&osVersion=98&productName=Camunda+Modeler&plugins%5Bid%5D=plugin1&plugins%5Bname%5D=plugin1&plugins%5Bid%5D=plugin2&plugins%5Bname%5D=plugin2');
```
Comment (barmac) [inline client/src/plugins/update-checks/__tests__/UpdateChecksSpec.js:325]: Is there any difference between `Camunda+Modeler` and `Camunda%20Modeler` as URI param?

Comment (oguzeroglu) [reply client/src/plugins/update-checks/__tests__/UpdateChecksSpec.js:325]: I googled this, they are kinda the same, implementation specific. Both URL safe. We use the URL API, so we're good there.

Comment (barmac) [reply client/src/plugins/update-checks/__tests__/UpdateChecksSpec.js:325]: I'm asking because of this:
image

Comment (barmac) [reply client/src/plugins/update-checks/__tests__/UpdateChecksSpec.js:325]: On the other hand, `URLSearchParams` use `+` here.

Let's leave it then :)

Comment (oguzeroglu) [reply client/src/plugins/update-checks/__tests__/UpdateChecksSpec.js:325]: I'll double check to make sure.

Comment (oguzeroglu) [reply client/src/plugins/update-checks/__tests__/UpdateChecksSpec.js:325]: 👍

Comment (oguzeroglu) [reply client/src/plugins/update-checks/__tests__/UpdateChecksSpec.js:325]: I made this request:
`"/update-check?editorID=813332dc-73c9-4e56-a05e-1e345ddd86d4&newerThan=v4.0.0&modelerVersion=v4.0.0&os=darwin&osVersion=18.7.0&productName=Camunda+Modeler"` and checked the /events data. We only have "Camunda Modeler" but no "Camunda+Modeler", so we're good to go 👍

Comment (barmac) [reply client/src/plugins/update-checks/__tests__/UpdateChecksSpec.js:325]: That's great.

File: client/src/plugins/update-checks/UpdateChecksAPI.js (hunk 1 above)
Comment (barmac) [reply client/src/plugins/update-checks/UpdateChecksAPI.js:18]: I think we're ready to merge the PR when you address this.

Comment (barmac) [reply client/src/plugins/update-checks/UpdateChecksAPI.js:18]: On the other hand, we use hard-coded values in other places: https://github.com/camunda/camunda-modeler/blob/c1b8a6d9e132e16d1f10d7e7c66709d2ef17d6f0/app/lib/index.js#L52

Meh.

Still, 
```suggestion
const PRODUCT_NAME = 'Camunda Modeler';
```
would make it more resilient.

Comment (oguzeroglu) [reply client/src/plugins/update-checks/UpdateChecksAPI.js]: I'm ok with injecting this: https://github.com/camunda/camunda-modeler/pull/1855/commits/3e666500a2f0bdbf74db7deee1e64ef66769e79a

Should we keep this commit?

Comment (oguzeroglu) [reply client/src/plugins/update-checks/UpdateChecksAPI.js]: ah yeah, let me update the var name

Comment (oguzeroglu) [reply client/src/plugins/update-checks/UpdateChecksAPI.js]: Done -> https://github.com/camunda/camunda-modeler/pull/1855/commits/149955b72047fa09f9b0fae46621f7efd914cb0a

**Comment by barmac** [comment]
Last thing: Can you please squash https://github.com/camunda/camunda-modeler/pull/1855/commits/149955b72047fa09f9b0fae46621f7efd914cb0a so that we keep clean commit history?

**Comment by oguzeroglu** [comment]
Done 👍

**Pull request merged**